### PR TITLE
Make mongodb_settings mandatory if cyral_repository.type = mongodb

### PR DESCRIPTION
**Pull request by wcmjunior**
## Description of the change

Fix a bug that allows empty blocks for `cyral_repository. mongodb_settings` when `cyral_repository.type = mongodb`.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

Acceptance tests were executed against `stable` and all tests passed.

### CODE REVIEW

File: cyral/resource_cyral_repository.go [hunk 1]
```diff
@@ -114,7 +136,13 @@ func (r *RepoInfo) ReadFromSchema(d *schema.ResourceData) error {
...
 	r.RepoNodesFromInterface(d.Get(RepoNodesKey).([]interface{}))
 	r.ConnDrainingFromInterface(d.Get(RepoConnDrainingKey).(*schema.Set).List())
-	r.MongoDBSettingsFromInterface(d.Get(RepoMongoDBSettingsKey).(*schema.Set).List())
+	var mongoDBSettings = d.Get(RepoMongoDBSettingsKey).(*schema.Set).List()
+	if r.Type == MongoDB && (mongoDBSettings == nil || len(mongoDBSettings) == 0) {
+		return fmt.Errorf("'%s' block is mandatory when 'type=%s'", RepoMongoDBSettingsKey, MongoDB)
+	} else if r.Type != MongoDB && len(mongoDBSettings) > 0 {
+		return fmt.Errorf("'%s' block is only allowed when 'type=%s'", RepoMongoDBSettingsKey, MongoDB)
+	}
+	r.MongoDBSettingsFromInterface(mongoDBSettings)
```
Comment (ricardorey10) [inline cyral/resource_cyral_repository.go:145]: Within `MongoDBSettingsFromInterface()` there's still a missing check: if the server type is `replicaset`, then the replica set name MUST be supplied. If it's not, then the API will complain, so nothing terrible will happen, but we should probably fail earlier.

File: docs/resources/repository.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ resource "cyral_repository" "minimal_repo" {
         host = "mongodb.cyral.com"
         port = 27017
     }
+
+    mongodb_settings {
```
Comment (ricardorey10) [inline docs/resources/repository.md:24]: Should we maybe add an example of a replica set repo?

**Review by ricardorey10 — APPROVED** [review]
LGTM modulo a missing validation in `MongoDBSettingsFromInterface()`. I'll approve now so that you can merge as soon as you address that.

File: docs/resources/repository.md (hunk 2 above)
Comment (wcmjunior) [reply docs/resources/repository.md:24]: We do have. Please check the resource that starts in line `51`. See: https://github.com/cyralinc/terraform-provider-cyral/blame/107228b3fcbac6a59efebf4474e8e521052592a6/docs/resources/repository.md#L51-L77

File: cyral/resource_cyral_repository.go (hunk 1 above)
Comment (wcmjunior) [reply cyral/resource_cyral_repository.go:145]: Perfect. Will add this check. Thanks!

File: docs/resources/repository.md (hunk 2 above)
Comment (ricardorey10) [reply docs/resources/repository.md:24]: Gotcha, thanks for sharing that link :grin:

**Review by VictorGFM — APPROVED** [review]
LGTM!

**Pull request merged**